File: triplicate/football/football.py
#!/usr/bin/env python3

num = input()

zeroes=0
ones=0
res=False
for c in num:
    if int(c) == 0:
        zeroes += 1
        ones = 0
    else:
        ones += 1
        zeroes = 0
    if zeroes >= 7 or ones >= 7:
        res=True

if res:
    print("YES")
else:
    print("NO")

File: cpp/football.cpp
#include <bits/stdc++.h>

auto main() -> int {
  std::string num;
  std::cin >> num;

  int zeroes = 0; int ones = 0; bool res = false;

  for (int i = 0; i < num.length(); ++i) {
    if (num[i] == '0') {
      zeroes++;
      ones = 0;
    } else {
      ones++;
      zeroes = 0;
    }
    if (zeroes >= 7 || ones >= 7) {
      res = true;
    }
  }

  if (res) {
    std::cout << "YES\n";
  } else {
    std::cout << "NO\n";
  }
}
